### ISSUE TITLE
Bug in line aggregation for float ranges

### ISSUE THREAD

**Issue by philippjfr** [bokeh/datashader#342]
I haven't been able to track down the precise bug, but it seems when aggregating using the ``canvas.line`` the ranges are not computed correctly resulting in the lines being abruptly cut-off. Presumably this was introduced as part of the fix for another bug in https://github.com/bokeh/datashader/pull/330.

Here's a minimal example (and [here](http://philippjfr.com/ipynbs/cutoff.csv)'s the data):

```python
import pandas as pd
import datashader as ds
import datashader.transfer_functions as tf

%matplotlib inline 
df = pd.read_csv('cutoff.csv')

x_range = df.time.min(), df.time.max()
y_range = df.Voltage.min(), df.Voltage.max()
print("Time interval:", x_range)
print("Voltage interval:", y_range)
cvs = ds.Canvas(x_range=x_range, y_range=y_range, plot_height=400, plot_width=400)
agg = cvs.line(df, 'time', 'Voltage')
agg.plot.imshow()
```

('Time interval:', (5.0, 7.0))
('Voltage interval:', (17.226610768, 22.066539451500002))

![image](https://cloud.githubusercontent.com/assets/1550771/25667427/50d78cda-301b-11e7-988a-29f5fa9d76bf.png)

You can see that even though we're aggregating over the entire interval, the curves are cut-off half-way.

**Comment by philippjfr** [comment]
Thinking about it, the bug it seems to be that it's treating the data as integers when testing for inclusivity. Any value above 6.0 is treated as 7.0, which is outside the exclusive bounds.

**Comment by jbednar** [comment]
Good observation; thanks.  @jbcrail?

**Comment by jbcrail** [comment]
@philippjfr When I fixed the previous segfault in #330, I decremented both upper bounds by one. This caused the clipping behavior in your test case.

This worked for the tests since we use integer bounds, but fails for float bounds. I'll have to figure out how to account for the exclusive bound while working for both integer and float ranges.

**Comment by jbcrail** [comment]
@philippjfr @jbednar I think I finally figured out the proper solution to the off-by-one issue. PR #343 moves the exclusive upper bound check down one level. It now matches the point aggregation more closely.

Here's the output from the above example with the PR applied:

![image](https://cloud.githubusercontent.com/assets/6038/25708561/d981d330-30b4-11e7-8905-58d45780142b.png)
